test/e2e/data/cluster-templates/docker-rke2.yaml

```
@@ -71,13 +73,19 @@ metadata:
 spec:
   template:
     spec:
-      customImage: kindest/node:v1.28.0
+      customImage: kindest/node:${KUBERNETES_VERSION}
       bootstrapTimeout: 15m
 ---
 apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
 kind: RKE2ConfigTemplate
 metadata:
   name: ${CLUSTER_NAME}-md-0
+spec: 
+  template:
+    spec:
+      agentConfig:
+        nodeAnnotations:
+          test: "true"
```

Node annotations is a specific feature, I don’t see it making sense to duplicate tests from RKE2 provider previously implemented here
```suggestion
spec: {}
```